project/Build.scala

```diff
@@ -1105,9 +1105,12 @@ object Build {
             -- "ObjectTest.scala" // compile errors caused by #9588
             -- "StackTraceTest.scala" // would require `npm install source-map-support`
             -- "UnionTypeTest.scala" // requires the Scala 2 macro defined in Typechecking*.scala
+            -- "PromiseMock.scala" -- "AsyncTest.scala" // TODO: Enable once we use a Scala.js with https://github.com/scala-js/scala-js/pull/4451 in
```

Comment:
Perhaps, for `PromiseMock.scala`, you could copy-paste the one from scala-js master into `tests/sjs-junit/test/org/scalajs/testsuite/jsinterop/`, and un-blacklist `AsyncTest.scala`. Once we upgrade, we would remove that copy and re-enable the original.

The main reason I'm suggesting that is that `PromiseMock.scala` happens to be the file with the most "real world" usage of union types in the Scala.js codebase. So it contains a lot of more complex uses of APIs that take and return values of union types. It would be good to know that it keeps working under the new scheme.